.github/workflows/main.yml

```diff
@@ -0,0 +1,24 @@
+name: Firmware Build
+run-name: ${{ github.actor }} is building your firmware 🚀
+on: [push, workflow_dispatch]
+jobs:
+  build-firmware:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
+      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by GitHub!"
+      - run: echo "🔎 The name of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
+      - name: Check out repository code
+        uses: actions/checkout@v3
+      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
+      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
+      - name: Test Build
+        uses: addnab/docker-run-action@v3
+        with:
+          image: quinn1876/waterloop-firmware:latest
```

Comment:
Where is this docker image? Is it a package in your github account?

Maybe it would be better to have this on the waterloop github? IDK how/if docker images can be stored on Github, so some elaboration on this would be nice.